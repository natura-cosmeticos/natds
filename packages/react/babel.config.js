const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
