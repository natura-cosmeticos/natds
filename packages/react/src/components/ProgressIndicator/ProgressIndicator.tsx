import React from 'react'
import { ProgressIndicatorProps } from './ProgressIndicator.props'
import styles from './ProgressIndicator.styles'

const ProgressIndicator = ({
  showLayer = false,
  size = 'medium',
  testID = 'ds-progress-indicator',
  variant = 'circular'
}: ProgressIndicatorProps): JSX.Element => {
  const { base, layer } = styles({ size, showLayer, variant })

  return (
    <div className={layer} role="progressbar" data-testid={testID}>
      <div className={base} />
    </div>
  )
}

export default ProgressIndicator
