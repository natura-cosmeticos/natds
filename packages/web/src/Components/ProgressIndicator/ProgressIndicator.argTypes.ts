/* eslint-disable no-magic-numbers */
export const argTypes = {
  className: {
    control: { type: null },
    description: "Override or extend the styles applied to the component.",
    table: { type: { summary: "object" } },
  },
  color: {
    control: {
      type: "select",
      options: ["primary", "secondary", "inherit"],
    },
    description:
      "The color of the component. It supports those theme colors that make sense for this component.",
  },
  size: {
    control: {
      type: "select",
      options: [24, 32, 40, 48],
    },
    description: "The size of the circle.",
    table: { type: { summary: "number" } },
  },
  disableShrink: {
    table: {
      defaultValue: { summary: false },
      type: { summary: "bool" },
    },
    description: "If true, the shrink animation is disabled.",
  },
};

export default argTypes;
