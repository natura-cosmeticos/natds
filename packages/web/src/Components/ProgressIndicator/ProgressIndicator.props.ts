/* eslint-disable no-magic-numbers */
export interface IProgressIndicatorProps {

  /**
   * Override or extend the styles applied to the component.
   *
   * @optional
   * @type object
   */
  className?: string;

  /**
   * If `true`, a layer will be applied behind the progress indicator
   *
   * @default false
   * @optional
   * @type boolean
   */
  showLayer?: boolean;

  /**
   * The size of the circle.
   *
   * @default 48
   * @optional
   * @type number
   */
  size?: 24 | 32 | 48 | 64;
}
