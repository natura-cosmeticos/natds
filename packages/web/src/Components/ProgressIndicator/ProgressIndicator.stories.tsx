import React from "react";
import { Meta, Story } from "@storybook/react";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import { ProgressIndicator } from "./ProgressIndicator";
import { IProgressIndicatorProps } from "./ProgressIndicator.props";
import { argTypes } from "./ProgressIndicator.argTypes";

export default {
  argTypes,
  component: ProgressIndicator,
  title: "Components/Progress Indicator",
} as Meta;

const useStyles = makeStyles(
  createStyles({
    progressIndicator: {
      margin: "0 16px",
    },
  }),
  { name: "ProgressIndicatorDemo" },
);

ProgressIndicator.displayName = "ProgressIndicator";

export const Template: Story<IProgressIndicatorProps> = (
  args: IProgressIndicatorProps) => <ProgressIndicator {...args} />;

export const Colors: Story<IProgressIndicatorProps> = (args) => {
  const classes = useStyles();

  return (
    <>
      <ProgressIndicator
        {...args}
        color="primary"
        className={classes.progressIndicator}
        disableShrink={true}
      />
      <ProgressIndicator
        {...args}
        color="secondary"
        className={classes.progressIndicator}
        disableShrink={true}
      />
      <ProgressIndicator
        {...args}
        color="inherit"
        className={classes.progressIndicator}
        disableShrink={true}
      />
    </>
  );
};

export const Sizes: Story<IProgressIndicatorProps> = (args) => {
  const classes = useStyles();

  return (
    <>
      <ProgressIndicator
        {...args}
        size={24}
        className={classes.progressIndicator}
        disableShrink={true}
      />
      <ProgressIndicator
        {...args}
        size={32}
        className={classes.progressIndicator}
        disableShrink={true}
      />
      <ProgressIndicator
        {...args}
        size={40}
        className={classes.progressIndicator}
        disableShrink={true}
      />
      <ProgressIndicator
        {...args}
        size={48}
        className={classes.progressIndicator}
        disableShrink={true}
      />
    </>
  );
};
