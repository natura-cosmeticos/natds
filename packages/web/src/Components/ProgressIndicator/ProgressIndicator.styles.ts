/* eslint-disable no-magic-numbers */
import createStyles from '@material-ui/core/styles/createStyles'
import { makeStyles } from '@material-ui/core'
import { IProgressIndicatorProps } from './ProgressIndicator.props'
import { IThemeWeb } from '../../Themes'

const defaultSize = 48

export const getLayerSize = ({ size }: IProgressIndicatorProps): number => (size ? size + 4 : defaultSize + 4)

const style = (theme :IThemeWeb) => createStyles({
  circularLayer: {
    backgroundColor: theme.palette?.background?.paper,
    height: getLayerSize,
    width: getLayerSize,
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  }
})

export const useStyles = makeStyles(style, { name: 'NatdDSProgressIndicator' })

export default useStyles
