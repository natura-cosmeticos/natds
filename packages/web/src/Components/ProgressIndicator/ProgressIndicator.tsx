/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import MaterialCircularProgress from "@material-ui/core/CircularProgress";
import { IProgressIndicatorProps } from "./ProgressIndicator.props";
import { useStyles } from "./ProgressIndicator.styles";

/**
 * ```
 * import { ProgressIndicator } from "@naturacosmeticos/natds-web"
 * ```
 */

export const ProgressIndicator = ({ layer = false, ...props }: IProgressIndicatorProps) => {
  const { circularLayer } = useStyles(props);

  return (
    <>
      {layer
        ? <div className={circularLayer}>
          <MaterialCircularProgress {...props} />
        </div>
        : <MaterialCircularProgress {...props} />
      }
    </>
  );
};

ProgressIndicator.displayName = "ProgressIndicator";

/**
 * @deprecated `CircularProgress` component will be deprecated soon.
 * Use `ProgressIndicator` instead
 */
export const CircularProgress = ProgressIndicator;

export default { CircularProgress, ProgressIndicator };
