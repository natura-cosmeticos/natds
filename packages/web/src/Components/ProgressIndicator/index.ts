import { CircularProgress, ProgressIndicator } from './ProgressIndicator'

export { IProgressIndicatorProps } from './ProgressIndicator.props'
export { CircularProgress, ProgressIndicator }
export default ProgressIndicator
