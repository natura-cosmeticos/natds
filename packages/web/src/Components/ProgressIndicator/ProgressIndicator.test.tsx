import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { ProgressIndicator } from "./ProgressIndicator";

describe("ProgressIndicator components", () => {
  it("should match to snapshot - default props", () => {
    const testRenderer = TestRenderer.create(<ProgressIndicator />);

    expect(testRenderer).toMatchSnapshot();
  });

  it("should match to snapshot - with size 24", () => {
    const testRenderer = TestRenderer.create(<ProgressIndicator size={24} />);

    expect(testRenderer).toMatchSnapshot();
  });

  it("should match to snapshot - with layer", () => {
    const testRenderer = TestRenderer.create(<ProgressIndicator showLayer={true} />);

    expect(testRenderer).toMatchSnapshot();
  });

  it("should match to snapshot - with size 32", () => {
    const testRenderer = TestRenderer.create(<ProgressIndicator size={32} />);

    expect(testRenderer).toMatchSnapshot();
  });

  it("should match to snapshot - with size 64", () => {
    const testRenderer = TestRenderer.create(<ProgressIndicator size={64} />);

    expect(testRenderer).toMatchSnapshot();
  });
});
