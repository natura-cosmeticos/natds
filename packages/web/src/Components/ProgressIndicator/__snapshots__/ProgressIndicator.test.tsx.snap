// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator components should match to snapshot - default props 1`] = `
<div
  className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`ProgressIndicator components should match to snapshot - with layer 1`] = `
<div
  className="NatdDSProgressIndicator-circularLayer-1 NatdDSProgressIndicator-circularLayer-4"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`ProgressIndicator components should match to snapshot - with size 24 1`] = `
<div
  className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 24,
      "width": 24,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`ProgressIndicator components should match to snapshot - with size 32 1`] = `
<div
  className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 32,
      "width": 32,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`ProgressIndicator components should match to snapshot - with size 64 1`] = `
<div
  className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 64,
      "width": 64,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;
